edgelet/iotedged/src/windows.rs

```diff
@@ -44,21 +45,18 @@ fn iotedge_service_main(args: Vec<OsString>) {
 }
 
 fn run_as_service(_: Vec<OsString>) -> Result<ServiceStatusHandle, Error> {
+    // Configure a Signal Future to alert us if Windows shuts us down.
+    let windows_signal = signal_future::signal();
+    let ws_signaler = RefCell::new(windows_signal.clone());
```

Comment:
Why did we need a `RefCell`? Couldn't `ws_signaler` be `mut`? For e.g.:
```rust
let mut ws_signaler = windows_signal.clone();
```

Reply:
Yeah we should prefer `mut` over a `RefCell` if possible.

Reply:
You need the `RefCell` because the `register` callback is `Fn`. That's why the original code before #1113 also used it.